config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/beans/JmsDestination.java

```diff
@@ -0,0 +1,366 @@
+        this.inboundDetail = inboundDetail;
+    }
+
+    public OutboundJmsDestinationDetail getOutboundDetail() {
+        return outboundDetail;
+    }
+
+    public void setOutboundDetail(OutboundJmsDestinationDetail outboundDetail) {
+        this.outboundDetail = outboundDetail;
+    }
+
+    public Map<String, Object> getAdditionalProperties() {
+        return additionalProperties;
+    }
+
+    public void setAdditionalProperties(Map<String, Object> additionalProperties) {
+        this.additionalProperties = additionalProperties;
+    }
+
+    @Override
+    public void postLoad(String entityKey, Bundle bundle, @Nullable File rootFolder, IdGenerator idGenerator) {
+        if (getJndiPasswordRef() != null && getJndiPassword() != null) {
+            throw new ConfigLoadException("Cannot specify both a password reference and a password for JNDI password for JMS destination: " + entityKey);
+        }
+
+        if (getDestinationPasswordRef() != null && getDestinationPassword() != null) {
+            throw new ConfigLoadException("Cannot specify both a password reference and a password for Destination password for JMS destination: " + entityKey);
+        }
+
+        if (this.inboundDetail == null && this.outboundDetail == null) {
+            throw new ConfigLoadException("Must specify inbound or outbound details for JMS destination: " + entityKey);
+        }
+        
+        if (this.inboundDetail != null && this.outboundDetail != null) {
+            throw new ConfigLoadException("Cannot specify both an inbound and an outbound details for JMS destination: " + entityKey);
+        }
+    }
+    
+    public enum DestinationType {
+        QUEUE("Queue"),
+        TOPIC("Topic");
+        
+        private String type;
+        
+        DestinationType(String type) {
+            this.type = type;
+        }
+        
+        public String getType() {
+            return type;
+        }
+        
+        public static DestinationType fromType(String type) {
+            return stream(values()).filter(c -> c.type.equals(type)).findFirst().orElse(null);
+        }
+    }
+    
+    public static class Builder {
+        private String name;
+        private String id;
+        private String providerType;
+        private String initialContextFactoryClassName;
+        private String jndiUrl;
+        private String jndiUsername;
+        private String jndiPasswordRef;
+        private String jndiPassword;
+        private Map<String, Object> jndiProperties;
+        private DestinationType destinationType;
+        private String connectionFactoryName;
+        private String destinationName;
+        private String destinationUsername;
+        private String destinationPasswordRef;
+        private String destinationPassword;
+        private InboundJmsDestinationDetail inboundDetail;
+        private OutboundJmsDestinationDetail outboundDetail;
+        private Map<String, Object> additionalProperties;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove the "additionalProperties" field and declare it as a local variable in the relevant methods. ([squid:S1450](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AS1450))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=138&issues=AWelBKp7XtJ12O8ukCjz&open=AWelBKp7XtJ12O8ukCjz)

[ISSUE_KEY]: AWelBKp7XtJ12O8ukCjz